pipeline/titanic/modeling_code.py

```diff
@@ -0,0 +1,94 @@
+
+from pyspark.ml.feature import (
+    VectorAssembler,
+    StringIndexer
+)
+
+TRAINING_DF_INDEX = 0
+TESTING_DF_INDEX = 1
+
+training_df = training_df.withColumnRenamed('Survived', 'label')
+testing_df = testing_df.withColumn('label', lit(0))
+datasets_list = [training_df, testing_df]
+
+for index, dataset in enumerate(datasets_list):
+    dataset = dataset.withColumn(
+        "Initial",
+        regexp_extract(col("Name"), "([A-Za-z]+)\.", 1))
+    datasets_list[index] = dataset
+
+misspelled_initials = [
+    'Mlle', 'Mme', 'Ms', 'Dr',
+    'Major', 'Lady', 'Countess',
+    'Jonkheer', 'Col', 'Rev',
+    'Capt', 'Sir', 'Don'
+]
+correct_initials = [
+    'Miss', 'Miss', 'Miss', 'Mr',
+    'Mr', 'Mrs', 'Mrs',
+    'Other', 'Other', 'Other',
+    'Mr', 'Mr', 'Mr'
+]
+for index, dataset in enumerate(datasets_list):
+    dataset = dataset.replace(misspelled_initials, correct_initials)
+    datasets_list[index] = dataset
+
+initials_age = {"Miss": 22,
+                "Other": 46,
+                "Master": 5,
+                "Mr": 33,
+                "Mrs": 36}
+for index, dataset in enumerate(datasets_list):
+    for initial, initial_age in initials_age.items():
+        dataset = dataset.withColumn(
+            "Age",
+            when((dataset["Initial"] == initial) &
+                 (dataset["Age"].isNull()), initial_age).otherwise(
+                dataset["Age"]))
+        datasets_list[index] = dataset
+
+for index, dataset in enumerate(datasets_list):
+    dataset = dataset.na.fill({"Embarked": 'S'})
+    datasets_list[index] = dataset
+
+for index, dataset in enumerate(datasets_list):
+    dataset = dataset.withColumn("Family_Size", col('SibSp') + col('Parch'))
+    dataset = dataset.withColumn('Alone', lit(0))
+    dataset = dataset.withColumn(
+        "Alone",
+        when(dataset["Family_Size"] == 0, 1).otherwise(dataset["Alone"]))
+    datasets_list[index] = dataset
+
+text_fields = ["Sex", "Embarked", "Initial"]
+for column in text_fields:
+    for index, dataset in enumerate(datasets_list):
+        dataset = StringIndexer(
+            inputCol=column, outputCol=column + "_index"). \
+            fit(dataset). \
+            transform(dataset)
+        datasets_list[index] = dataset
+
+non_required_columns = ["Name", "Embarked", "Sex", "Initial"]
+for index, dataset in enumerate(datasets_list):
+    dataset = dataset.drop(*non_required_columns)
+    datasets_list[index] = dataset
+
+training_df = datasets_list[TRAINING_DF_INDEX]
+testing_df = datasets_list[TESTING_DF_INDEX]
+
+columns_without_label = training_df.columns.copy()
+columns_without_label.remove("label")
+
+assembler = VectorAssembler(
+    inputCols=columns_without_label,
+    outputCol="features")
+assembler.setHandleInvalid('skip')
+
+features_training = assembler.transform(training_df)
+(features_training, features_evaluation) = \
+    features_training.randomSplit([0.8, 0.2], seed=33)
+features_testing = assembler.transform(testing_df)
```

Comment:
The last comment, do you change the maximum column value in pycharm from 120 to 80? this config follows the PEP8 specification, case you don't change this config, please make the change and runs again the pycharm code formatting to all files altered.